### PR TITLE
Add typehints and modules to Rust code

### PR DESCRIPTION
**Pull request by ayyaruq**
**Purpose**
Initially I wanted to just add type hints. Scope creep happened and now it's also renaming the dir as well as setting up fake python modules. `rustfmt` was also run so basically the diff is huge, recommend looking at specific commits.

For the module layout, I chose to do nested modules inline, but there's no reason they couldn't be registered distinctly and then connected later. The hack comment is to force it to present as real packages to simplify imports on the python side. Could use folders in rust too but I see little benefit to it at this point.

**New Features**
- python can now see type hints for rust functions
- python can import with submodules (`from nari_act_ext.utils import get_time_milliseconds`), allowing python to stub the same structure for fallback

**Before/After**
Before: `act/ext/`
After: `act/rust/`

**Additional Context**
Might help to look at this as individual commits or sets of, since the diff is full-file due to dir rename and `rustfmt`.

### CODE REVIEW

File: act/rust/lines_to_params.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
+use crate::actor;
+use crate::parser;
+use crate::utils;
+use pyo3::prelude::*;
+
+/// Params to ability
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[any]")]
```
Comment (wolfcomp) [inline act/rust/lines_to_params.rs:8]: This should have a return type of `tuple[tuple[int,str],list[int],list[float],tuple[int,str],list[int],list[float],list[tuple[int,int,int,int,int,int,int]],int]`

File: act/rust/lines_to_params.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            .chunks(2)
+            .map(|x| action_effect_from_params(x.to_vec()))
+            .collect::<Vec<(u8, u8, u8, u8, u16, u8, u8)>>(),
+        sequence,
+    )
+}
+
+/// Params to action_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[int]")]
```
Comment (wolfcomp) [inline act/rust/lines_to_params.rs:62]: This should be `tuple[int,int,int,int,int,int,int]`

File: act/rust/lines_to_params.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        param0,
+        param1,
+        parser::param_to_4_byte_float(inp.get(1).unwrap()),
+        parser::param_to_4_byte_int(inp.get(2).unwrap()),
+    )
+}
+
+/// Params to statuslist
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[any]")]
```
Comment (wolfcomp) [inline act/rust/lines_to_params.rs:90]: This should be return type `tuple[tuple[int,str],str,list[int],list[float],list[tuple[int,int,float,int]]]`

File: act/rust/parser.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+/// Param to 4-byte integer
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> int")]
+pub(crate) fn param_to_4_byte_int(inp: &str) -> u32 {
+    u32::from_str_radix(inp, 16).unwrap()
+}
+
+/// Param to four 1-byte integers
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[int]")]
```
Comment (wolfcomp) [inline act/rust/parser.rs:37]: This should be return type `tuple[int,int,int,int]`

File: act/rust/lib.rs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    m.add_submodule(parser_module)?;
+
+    let utils_module = PyModule::new(py, "utils")?;
+    utils_module.add_function(wrap_pyfunction!(utils::date_from_act_timestamp, m)?)?;
+    utils_module.add_function(wrap_pyfunction!(utils::pad4, m)?)?;
+    utils_module.add_function(wrap_pyfunction!(utils::pad8, m)?)?;
+    utils_module.add_function(wrap_pyfunction!(utils::validate_checksum, m)?)?;
+    m.add_submodule(utils_module)?;
+
+    // HACK: abuse python imports to make `from rustext.utils import validate_checksum` work
```
Comment (nonowazu) [inline act/rust/lib.rs:42]: could we add some more docs for this, just in case there's a better way that comes out later.

**Review by nonowazu — CHANGES_REQUESTED** [review]

**Review by nonowazu — APPROVED** [review]

**Pull request merged**